Modify the extensions that show up in "Installed marketing extensions" card

nima-karimi opened the pull request:
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes #34431.

This PR adds seven new extensions to the "Installed marketing extensions" list on the `Marketing > Overview` page. These extensions appear on the list when the user installs and activates them. The added extensions are:

- [AutomateWoo Referrals](https://woocommerce.com/products/automatewoo-refer-a-friend/)
- [AutomateWoo Birthdays](https://woocommerce.com/products/automatewoo-birthdays/)
- [Creative Mail by Constant Contact](https://woocommerce.com/products/creative-mail-for-woocommerce/)
- [Jetpack CRM](https://woocommerce.com/products/jetpack-crm/)
- [Salesforce](https://woocommerce.com/products/integration-with-salesforce-for-woocommerce/)
- [Tiktok](https://woocommerce.com/products/tiktok-for-woocommerce/)
- [Zapier](https://woocommerce.com/products/woocommerce-zapier/)
- [Vimeo for WooCommerce](https://woocommerce.com/products/trustpilot-reviews/)
- [Trustpilot Reviews](https://woocommerce.com/products/trustpilot-reviews/)

Moreover, I have removed the [HubSpot](https://woocommerce.com/products/hubspot-for-woocommerce/) extension from the list based on the requirements defined in #34431.

This PR is based on Gan's PR #34415 and uses the extension icons introduced there.

Note: The `InstalledExtensions.php` file needed some cleanup/refactoring; however, I did not modify it because we will implement a new method for getting this data in the Multi-channel feature (pe2C5g-dl-p2) that will deprecate this class.

### How to test the changes in this Pull Request:

- General tests:

  1. Go to `WooCommerce Settings > Advanced > Features` and turn off the multichannel marketing feature.
  2. Install the seven extensions mentioned above
  3. Install AutomateWoo
  4. Go to the `Marketing > Overview` page and confirm their names appear in the "Installed marketing extensions" section
  5. Confirm that the "Get Support" and "Docs" links are correct and point to each extension's support page and documentation

- For [AutomateWoo Referrals](https://woocommerce.com/products/automatewoo-refer-a-friend/), [AutomateWoo Birthdays](https://woocommerce.com/products/automatewoo-birthdays/), [Jetpack CRM](https://woocommerce.com/products/jetpack-crm/), and [Zapier](https://woocommerce.com/products/woocommerce-zapier/):

  1. Confirm that the "Settings" link works and redirects the user to the extension's settings page.

- For [Creative Mail by Constant Contact](https://woocommerce.com/products/creative-mail-for-woocommerce/), [Salesforce](https://woocommerce.com/products/integration-with-salesforce-for-woocommerce/), [Vimeo for WooCommerce](https://woocommerce.com/products/trustpilot-reviews/), and [Tiktok](https://woocommerce.com/products/tiktok-for-woocommerce/):

  1. Ensure you have not set up the extension or completed onboarding. If you have, reset the extension data.
  2. Confirm that a link to "Finish setup" is displayed next to each extension name if it has not been set up and links to their onboarding/setup process.
  3. Once you complete setup/onboarding, confirm that the "Finish setup" link is replaced with the "Settings" link and points to the extension's settings page.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you successfully run tests with your changes locally?
-   [ ] Have you created a changelog file for each project being changed, i.e.` pnpm changelog add --filter=<project>`?

<!-- Mark completed items with an [x] -->

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

Commit:
Add new extensions to list of Installed Extensions on Marketing page

Commit:
Remove HubSpot from list of Installed Extensions on Marketing page

Commit:
Fix style

Commit:
Add Vimeo to list of installed marketing extensions

Commit:
Add Trustpilot to list of installed marketing extensions

Commit:
Fix Vimeo icon path

chihsuan commented on plugins/woocommerce/src/Admin/Marketing/InstalledExtensions.php:
```diff
@@ -319,6 +377,200 @@ protected static function get_mailpoet_extension_data() {
+
+			$data['settingsUrl'] = admin_url( 'admin.php?page=tiktok' );
+			$data['docsUrl']     = 'https://woocommerce.com/document/tiktok-for-woocommerce/';
+			$data['supportUrl']  = 'https://ads.tiktok.com/athena/user-feedback/?identify_key=6a1e079024806640c5e1e695d13db80949525168a052299b4970f9c99cb5ac78';
```
There is a system error when I click on the link. Is this expected?

https://user-images.githubusercontent.com/4344253/187343722-fb6c5b66-ac39-4a7c-bb98-b86ed1eb05b9.mov

nima-karimi replied:
Yes, that appears when you haven't logged in with your TikTok account. I agree that it's strange behavior and a bit confusing, but I'm using the same link as suggested in [TikTok's Documentation](https://woocommerce.com/document/tiktok-for-woocommerce/#section-9).

chihsuan commented:
> Confirm that the "Get Support" and "Docs" links are correct and point to each extension's support page and documentation

For other reviewers: The `multichannel-marketing` feature flag is enabled now. To test these links, I think we need to set `multichannel-marketing` to false in developement.json.

chihsuan commented on plugins/woocommerce/src/Admin/Marketing/InstalledExtensions.php:
```diff
@@ -319,6 +377,200 @@ protected static function get_mailpoet_extension_data() {
+
+			$data['settingsUrl'] = admin_url( 'admin.php?page=integration-with-salesforce' );
+			$data['docsUrl']     = 'https://woocommerce.com/document/salesforce-integration/';
+			$data['supportUrl']  = 'https://wpswings.com/submit-query/?utm_source=wpswings-salesforce-woo&utm_medium=woo-backend&utm_campaign=submit-query';
```
Is the query string needed?

 `?utm_source=wpswings-salesforce-woo&utm_medium=woo-backend&utm_campaign=submit-query`

nima-karimi replied:
You're right, we can remove that tracking URL. Done in https://github.com/woocommerce/woocommerce/pull/34444/commits/19b91307bc6e70d441919bc20e500adc7c314132

Commit:
Remove tracking parameters from URL

chihsuan commented:
Hi @nima-karimi Thanks for addressing the feedback! 

I just tested your PR, and found that if we don't activate `AutomateWoo`, `AutomateWoo - Refer A Friend Add-on` shows an **invalid** "Finish setup" link. Could you confirm it?


https://user-images.githubusercontent.com/4344253/188067720-6a80af66-7e56-48e0-b6ff-182a2bbf3790.mov

ecgan commented:
> For other reviewers: The `multichannel-marketing` feature flag is enabled now. To test these links, I think we need to set `multichannel-marketing` to false in developement.json.

Hi @chihsuan , the feature flag was developed in PR https://github.com/woocommerce/woocommerce/pull/33964. The flag is set to `true` based on https://github.com/woocommerce/woocommerce/blob/ca91250b2e17e88c902e460135c0531b3f632d90/plugins/woocommerce-admin/docs/features/feature-flags.md:

> Development - All features should be enabled in development.

As an alternative option to modifying the `development.json` file, you can also toggle the feature in the Settings > Advanced > Features page: `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`. PR https://github.com/woocommerce/woocommerce/pull/33964 contains a demo video on this. 🙂

Commit:
Show AW extensions docs and support link if AW not activated

If AutomateWoo extensions like "AutomateWoo - Refer a friend" is activated but AutomateWoo itself is not, we will not be able to link to their settings page. However, we can show links to their support page and documentation.

chihsuan reviewed (approved):
Thanks @nima-karimi tested well and looks nice! 👍
